### DISCUSSION TITLE
variable won't work when using script.sh file

### DISCUSSION THREAD

**Discussion by imed-ghomari** [Taitava/obsidian-shellcommands#200] [Q&A] [answered]
I'm on Mac, I'm using default (bash). The working directory is empty. I put my script in my vault's root.
I'm using the following shell command to call the script file: `/bin/bash script.sh`

I tried using `echo {{clipboard}}` or `echo {{file_path:relative}}` in the file.
None of the variable work, I always get just the text `{{clipboard}}` or `{{file_path:relative}}` in the notification balloon.
Am I doing something wrong ?

**Comment by Taitava** [comment, marked as answer]
Hi, thank you for your question! :)

`{{variables}}` only work when they are used in the shell comand text, i.e. in your case the shell command text is `/bin/bash script.sh`. The _Shell commands_ plugin does not read the `script.sh` file at all, it's only read by Bash, and so Bash does not know `{{clipboard}}` or `{{file_path:relative}}` variables, which are specific to the _Shell commands_ plugin.

What you can do to solve this, is to pass the values of these variables as _arguments_ to your script file. This is how it's done in Bash:

Your shell command: `/bin/bash script.sh {{clipboard}} {{file_path:relative}}`

An example content of `script.sh`:
```bash
echo $1 # Prints the value of {{clipboard}}, because {{clipboard}} is the first argument.
echo $2 # Prints the value of {{file_path:relative}}, because {{file_path:relative}} is the second argument.
```

- [x] I think I should mention this in the documentation. 🙂 Edit 2022-05-08: Here's the documentation: https://publish.obsidian.md/shellcommands/Variables/Variables+-+general+principles#Passing+variable+values+into+external+script+files
